cmd/initialize/cmd.go

```diff
@@ -140,7 +140,12 @@ func run(cmd *cobra.Command, argv []string) {
 		if strings.Contains(fmt.Sprintf("%s", err), "STS") {
 			r.OCMClient.LogEvent("ROSAInitCredentialsSTS", nil)
 		}
-		r.Reporter.Errorf("Error creating AWS client: %v", err)
+		if strings.Contains(fmt.Sprintf("%s", err), "InvalidClientTokenId") {
```

Comment:
@willkutler Rather than duplicating this logic throughout the code, how about a function in the `aws` package that handles client build errors? So _something_ like:

```
function HandleAWSClientBuildError(runtime *Runtime, err error) {
        if strings.Contains(fmt.Sprintf("%s", err), "STS") {
	r.OCMClient.LogEvent("ROSAInitCredentialsSTS", nil)
		}
		r.Reporter.Errorf("Error creating AWS client: %v", err)
		if strings.Contains(fmt.Sprintf("%s", err), "InvalidClientTokenId") {
			r.Reporter.Errorf(
				"Error creating AWS client: Invalid AWS Credentials. Update credentials in ~/.aws/credentials")
		} else {
			r.Reporter.Errorf("Error creating AWS client: %v", err)
		}
}
```

And then, where we fail to build the client:

```
if err != nil {
   aws.HandleAWSClientBuildError(r, err)
   os.Exit(1)
}
```

You could also write a unit test for the `HandleAWSClientBuildError` function 👍

Reply:
@willkutler Alternatively, can we hook this into the `build` method of the aws.ClientBuilder and wrap the `error` we return to have the friendly message about updating your AWS credentials? Following the code in the `build` method, we call `awsClient.ValidateCredentials()` via `getClientDetails()` function. This would mean that we have a single update and then anyone using the builder benefits from your new error message.

Reply:
moved logic to this function per your request and verified it works for all aws client build()

Reply:
moved logic to this function per your request and verified it works for all aws client build()